stream/src/main/scala/org/apache/pekko/stream/impl/fusing/Ops.scala

```diff
@@ -1402,8 +1402,7 @@ private[stream] object Collect {
     new GraphStageLogic(shape) with InHandler with OutHandler {
       override def toString = s"MapAsyncUnordered.Logic(inFlight=$inFlight, buffer=$buffer)"
 
-      private val decider =
-        inheritedAttributes.mandatoryAttribute[SupervisionStrategy].decider
+      private lazy val decider = inheritedAttributes.mandatoryAttribute[SupervisionStrategy].decider
```

Comment:
Can we make this a `def`? In most places in this source file, this is a `def`.

The `def` means no synchronize block. `lazy val`s come with the overhead of having `synchronize` in the generated byte code.

Reply:
1. `def` will do recaculation, and that should be avoid.
2. `lazy val` will only do synchronization the first time, and the supervision strategy method is expected to be quick for calculation, so even't the undering Thread is a VirtualThread, it will not cause any problem too.